### PR TITLE
Report  label of lists in browseMode

### LINKED ISSUE #7652: NVDA not announcing aria-label on lists
Steps to reproduce:

1. Go to https://codepen.io/Venkatesh-a11y/pen/OxZWdv
2. As soon as page loads focus lands on edit field so hit ESC key to switch to browse mode.
3. Hit M to move to the result frame.
4. Now arrow down to both unordered and ordered list over there.
5. Or hit L to navigate lists on the page.

Expected behavior:

NVDA should announce the aria-label before announcing the list item when focus enters list with quick key L or with down arrow otherwise context will miss for the users though specific context is described for each and every list with aria-label.
In the mentioned example, when user hits L NVDA should announce "list with two items here are the list of fruits I like bullet apples" and in the next hit of L it should announce "list with two items here are the list of vegetables I like one potato".

Actual Result:

NVDA doesn't announce the aria-label with quick key L or while arrowing down the list instead it announces the number of list items followed by the first list item directly.
In the mentioned example, NVDA announces "list with two items bullet Apples" and "list with two items one Potato" when focus enters lists.

System configuration:

NVDA Version: 2017.3, Installed

Other information:

OS version: Windows 10
Browser Versions: Firefox 56, Chrome 61, IE 11, Microsoft Edge 41
Tested with NVDA 2017.1 and 2017.2 also and the issue repro's.

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Fixes #7652 

### Summary of the issue:
Web authors can label ul/ol lists with aria-label or aria-labelledby. However, NVDA does not currently speak these labels when entering the list with quicknav, or arrowing (when the label does not appear anywhere else on the page).
 
### Description of how this pull request fixes the issue:
The information spoken when entering a list has now been generlized so that it includes  all the standard properties such as name, states, level etc, which were previously being completely ignored.

### Testing performed:
Using Firefox, Tested the testcase in #7652. NVDA announced the name of both lists.

### Known issues with pull request:
None.

### Change log entry:
Bug fixes:
* When navigating into a list on the web, NvDA will now report its label if the web author has provided one.

### CODE REVIEW

**Comment by michaelDCurran** [comment]
In the past, all of the above if/elifs returned. Now one of them (the 
lists one) does not return and needs to get to this point.

Perhaps I could move the if for the lists down to inside this general if 
and switch it back to being an elif. Might that make the flow more 
understandable?

File: source/speech.py [hunk 1]
```diff
@@ -1196,8 +1200,8 @@ def getControlFieldSpeech(attrs,ancestorAttrs,fieldType,formatConfig=None,extraD
 			+ (" %s" % stateText if stateText else "")
 			+ (" %s" % ariaCurrentText if ariaCurrent else ""))
 
-	# General cases
-	elif (
+	# General cases.
+	if (
```
[the remark this answers is not in the document]
Comment (feerrenrut) [reply source/speech.py]: > In the past, all of the above if/elifs returned. Now one of them (the
lists one) does not return and needs to get to this point.

Oh I missed the removal of the `return`. 

> Perhaps I could move the if for the lists down to inside this general if
and switch it back to being an elif. Might that make the flow more
understandable?

No, I think leave it as is. It's the conditions themselves that make this hard to understand. It's hard to know which components of the conditions are exclusive and not.

File: source/speech.py [hunk 2]
```diff
@@ -1166,12 +1166,16 @@ def getControlFieldSpeech(attrs,ancestorAttrs,fieldType,formatConfig=None,extraD
 	# speakStatesFirst: Speak the states before the role.
 	speakStatesFirst=role==controlTypes.ROLE_LINK
 
+	containerContainsText="" #: used for item counts for lists
+
 	# Determine what text to speak.
 	# Special cases
 	if speakEntry and childControlCount and fieldType=="start_addedToControlFieldStack" and role==controlTypes.ROLE_LIST and controlTypes.STATE_READONLY in states:
```
Comment (feerrenrut) [inline source/speech.py]: Is it still important to have `speakEntry` here? It is checked before this value `containerContainsText` is used in the "General" section.

**Comment by michaelDCurran** [comment]
True. We technically don't have to check speakEntry in that if check 
anymore.

**Comment by michaelDCurran** [comment]
@feerrenrut: I removed the check of the speakEntry variable. Does this look okay now?

File: source/speech.py [hunk 3]
```diff
@@ -1196,8 +1200,8 @@ def getControlFieldSpeech(attrs,ancestorAttrs,fieldType,formatConfig=None,extraD
 			+ (" %s" % stateText if stateText else "")
 			+ (" %s" % ariaCurrentText if ariaCurrent else ""))
 
-	# General cases
-	elif (
+	# General cases.
+	if (
 		(speakEntry and ((speakContentFirst and fieldType in ("end_relative","end_inControlFieldStack")) or (not speakContentFirst and fieldType in ("start_addedToControlFieldStack","start_relative"))))
 		or (speakWithinForLine and not speakContentFirst and not extraDetail and fieldType=="start_inControlFieldStack")
```
Comment (derekriemer) [inline source/speech.py]: It may not be worth doing, but the a & not b & not c part here can be simplified to a & not (b or c)

**Review by feerrenrut — DISMISSED** [review]

**Review by feerrenrut — APPROVED** [review]
Assuming the build passes, this is ready to merge.

**Pull request merged**